### ISSUE TITLE
Array Fields Not Supported

### ISSUE THREAD

**Issue by ricxsar** [ramiel/mongoose-aliasfield#9]
Hi. I have a schema definition below:
```
var routeSchema = new Schema({
  _terminals    : [{ type: Schema.Types.ObjectId, ref: 'Terminal', alias: 'terminals' }],  
  _devices_activated       : [{ type: Schema.Types.ObjectId, ref: 'DeviceActivated', alias: 'devices' }], 
  _organization : { type: Schema.Types.ObjectId, ref: 'Organization', required: true, index: true, alias: 'organization' } 
}, { 
  timestamps : 1 
});
```
When converting document to alias object using `toAliasedFieldsObject()`,  the `_terminals` and `_devices_activated` are not transformed as `terminals` and `devices` respectively.

**Comment by ricxsar** [comment]
I have a workaround in this issue by using the below schema definition:
```
var routeSchema = new Schema({
  _terminals    : { type: [Schema.Types.ObjectId], ref: 'Terminal', alias: 'terminals' },  
  _devices_activated       : { type: [Schema.Types.ObjectId], ref: 'DeviceActivated', alias: 'devices' },  
  _organization : { type: Schema.Types.ObjectId, ref: 'Organization', required: true, index: true, alias: 'organization' }  
}, { 
  timestamps : 1 
});
```
But it breaks the `mongoose populate` functionality which is bad and some plugins like `mongoose-id-validator`

**Comment by ramiel** [comment]
Hi @ricxsar, I'm going to write some test for this and to correct it in the next days.
Thank you for reporting

**Comment by ramiel** [comment]
Ok, should be fixed in version 0.3.0
